consensus/propagation/commitment.go

```diff
@@ -144,7 +153,15 @@ func chunkToPartMetaData(chunk *bits.BitArray, partSet *proptypes.CombinedPartSe
 func (blockProp *Reactor) handleCompactBlock(cb *proptypes.CompactBlock, peer p2p.ID, proposer bool) {
 	added := blockProp.AddProposal(cb)
 	if !added {
+		p := blockProp.getPeer(peer)
+		if p == nil {
+			return
+		}
+		p.consensusPeerState.SetHasProposal(&cb.Proposal)
 		return
+	} else if !proposer {
+		p := blockProp.getPeer(peer)
```

Comment:
I would add a if p != nil check to avoid nil pointer exceptions

Reply:
ahh shoot sorry missed this after the other instance